routes/v1.js

```diff
@@ -7,38 +7,85 @@ var base        = require('taskcluster-base');
 
+
 /**
  * API end-point for version v1/
  *
  * In this API implementation we shall assume the following context:
  * {
  *   publisher:         // Publisher created with exchanges.js
- *   workerType:        // Instance of data.WorkerType
+ *   WorkerType:        // Instance of data.WorkerType
  * }
  */
 var api = new base.API({
   title:          "AWS Provisioner API Documentation",
   description: [
-    "The AWS  provisioner, typically available at",
-    "`aws-provisioner.taskcluster.net`, is responsible for provisioning EC2",
-    "instances as tasks become become pending. To do this it monitors the",
-    "state of queues, EC2 instances, spot prices and other interesting",
-    "parameters."
-    // TODO: Write who the intended users are:
-    //        - tools for defining workerTypes
-    //        - tools for inspecting state of AWS
-    //        - hooks monitoring state
-    // TODO: Write what the AWS provisioner does, how the API works
-    //       just some introduction...
-    //       To quote a the Daleks: Explain..., Explain,.. Exterminate!
-    //                                                    (^disregard the that)
+    "The AWS Provisioner is responsible for provisioning instances on EC2 for use in",
+    "TaskCluster.  The provisioner maintains a set of worker configurations which",
+    "can be managed with an API that is typically available at",
+    "aws-provisioner.taskcluster.net.  This API can also perform basic instance",
+    "management tasks in addition to maintaining the internal state of worker type",
+    "configuration information.",
+    "",
+    "The Provisioner runs at a configurable interval.  Each iteration of the",
+    "provisioner fetches a current copy the state that the AWS EC2 api reports.  In",
+    "each iteration, we ask the Queue how many tasks are pending for that worker",
+    "type.  Based on the number of tasks pending and the scaling ratio, we may",
+    "submit requests for new instances.  We use pricing information, capacity and",
+    "utility factor information to decide which instance type in which region would",
+    "be the optimal configuration.",
+    "",
+    "Each EC2 instance type will declare a capacity and utility factor.  Capacity is",
+    "the number of tasks that a given machine is capable of running concurrently.",
+    "Utility factor is a relative measure of performance between two instance types.",
+    "We multiply the utility factor by the spot price to compare instance types and",
+    "regions when making the bidding choices.",
+    "",
   ].join('\n')
 });
 
-// Export api
+
+function errorHandler(err, res, workerType) {
+  console.error(err, err.stack);
+  switch(err.code) {
+    case 'ResourceNotFound':
+      return res.status(404).json({
+        message: workerType + ': ' + err.body.message.value,
+        error: {
+          workerType: workerType,
+          reason: err.code,
```

Comment:
It's a bad idea to expose error codes from azure, fact that we azure table storage is internal, consumers have no business knowing that.
Message should be human readable and explain `"workerType: " workerType + " wasn't found",

Note, we should formalize errors, there is a bug on the issue, feel free to tune in:
https://bugzilla.mozilla.org/show_bug.cgi?id=1141639